src/uwtools/api/config.py

```diff
@@ -90,7 +106,31 @@ def realize(
     dry_run: bool = False,
 ) -> bool:
     """
-    ???
+    Realize an output config based on an input config and an optional values-providing config.
+
+    If no input is specified, ``stdin`` is read. If no output is specified, ``stdout`` is written
+    to.
+
+    When a filename is available for an input or output, its format will be deduced from its
+    extension, if possible. This can be overriden by specifying the format explicitly, and it is
+    required to do so for reads from ``stdin`` or writes to ``stdout``, as no attempt is made to
+    deduce the format of streamed data.
+
+    A ``dict`` may also be provided as an input value.
+
+    If ``values_needed`` is ``True``, a report of values needed to realize the config is logged.
+
+    In ``dry_run`` mode, output is written to ``stderr``.
+
+    :param input_config: Input config file (``None`` or unspecified => read ``stdin``).
+    :param input_format: Format of the input config.
+    :param output_file: Output config file (``None`` or unspecified => write to ``stdout``).
+    :param output_format: Format of the output config.
+    :param values: Source of values used to modify input.
+    :param values_format: Format of values when sourced from file.
+    :param values_needed: Report complete, missing, and template values.
```

Comment:
I was thinking that as a part of @elcarpenter's work with modifying terminology from `values` to `supplemental` was going to apply in the config, but not the templater. It seems reversed here.

Reply:
I think you've got it right: We want to move away from the term "values" in config-realize context, and keep "values" in "template render" context. Since file is config-realize context, we'll need to update this docstring to follow @elcarpenterNOAA's updates when those are available. I've just kept everything as-is in-sync for now, so that CLI, API, and library code all match on this branch. I hope I'm answering the question/concern -- please let me know if not.

Oh, I just realized, maybe you're specifically referring to the presence of the word "template" here. That's interesting: It has nothing to do with `template render`, but the "template" terminology appears here (and in the library code this connects to) because Jinja2 is still in play. Maybe we want to call this something different, but I think that change should start in the library code and when be reflected here (and in the CLI code if it also needs it).